### PR TITLE
Ansaf-1246

### PR DESCRIPTION
**Pull request by ansaf-1246**
Please review

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,19 @@
...
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+      <Navbar />
+      <Routes>
+        <Route path="/" element={<BooksPage />} />
+        <Route path="/cart" element={<CartPage />} />
+      </Routes>
```
Comment (atmeshwar-classic) [inline src/App.tsx]: Extract this into another file inside src/router/Router.tsx

File: src/domain/books/BooksPage.tsx [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+  }, [Bookdata]);
+
   return (
-    <div>
-      Books Page
+    <div className="book-page-container">
+      <>{console.log(Bookdata)}</>
+      {Bookdata.length === 0 ? (
+        <div>Loading...</div>
+      ) : (
```
Comment (atmeshwar-classic) [inline src/domain/books/BooksPage.tsx]: We get a loading state from the booksSelector, use it instead.

File: src/domain/books/booksSlice.ts [hunk 3]
```diff
@@ -1,58 +1,42 @@
...
+        state.error = (action.payload as string) ?? action.error.message;
+      });
+  },
+});
 
 export const booksSliceReducer = booksSlice.reducer;
 export const { cleanupBooks } = booksSlice.actions;
-export const booksSelector = (state: RootState) => state.books;
\ No newline at end of file
+export const booksSelector = (state: RootState) => state.books.books;
```
Comment (atmeshwar-classic) [inline src/domain/books/booksSlice.ts]: The default selector should always return the entire state. However, if you want to customise the  output, then you can create a new one.

File: src/domain/cart/cartSlice.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    removeBookFromCart: (state, action: PayloadAction<string>) => {
+      state.cart = state.cart.filter((e) => e.id !== action.payload);
+    },
+  },
+});
+
+export const cartSliceReducer = cartSlice.reducer;
+export const { addBookToCart, removeBookFromCart } = cartSlice.actions;
+export const cartSelector = (state: RootState) => state.cart.cart;
```
Comment (atmeshwar-classic) [inline src/domain/cart/cartSlice.ts]: Please see the above comment

File: src/utils/priceTotal.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { CartItem } from "../domain/cart/types";
+
+export const totalPrice = (cart: CartItem[]) => {
+  return cart.reduce(
+    (accumulator, currentValue) => (accumulator += currentValue.price),
+    0
+  );
+};
```
Comment (atmeshwar-classic) [inline src/utils/priceTotal.ts]: Since this fn is specific to the cart, you should move it to the domains/cart directory

**Review by atmeshwar-classic — CHANGES_REQUESTED** [review]
Hi @ansaf-1246 Please look at the comments I have made.
Also, the books page should not scroll horizontally. Please fix it.

File: src/App.tsx (hunk 1 above)
Comment (ansaf-1246) [reply src/App.tsx]: Done

File: src/domain/books/BooksPage.tsx (hunk 2 above)
Comment (ansaf-1246) [reply src/domain/books/BooksPage.tsx]: Done

File: src/domain/books/booksSlice.ts (hunk 3 above)
Comment (ansaf-1246) [reply src/domain/books/booksSlice.ts]: Done

File: src/domain/cart/cartSlice.ts (hunk 4 above)
Comment (ansaf-1246) [reply src/domain/cart/cartSlice.ts]: Done

File: src/utils/priceTotal.ts (hunk 5 above)
Comment (ansaf-1246) [reply src/utils/priceTotal.ts]: Done

File: src/domain/cart/CartPage.tsx [hunk 6]
```diff
@@ -3,23 +3,24 @@ import { CartItem } from "./types";
 import { cartSelector } from "./cartSlice";
 import "./styles.css";
 import CartCard from "../../components/CartCard/CartCard";
-import { totalPrice } from "../../utils/priceTotal";
+import { priceTotal } from "./priceTotal";
 
 export const CartPage = (): JSX.Element => {
   const cartData = useSelector(cartSelector);
```
Comment (atmeshwar-classic) [inline src/domain/cart/CartPage.tsx:9]: Just destructure the obj and only take the values that you want.

File: src/domain/books/BooksPage.tsx [hunk 7]
```diff
@@ -11,20 +11,23 @@ export const BooksPage = (): JSX.Element => {
...
-      )}
-    </div>
-  );
+      );
+    } else if (Bookdata.loading === "loading") {
+      return <div>{Bookdata.loading}...</div>;
+    } else {
+      return <div>An error occured...</div>;
+    }
+  };
```
Comment (atmeshwar-classic) [inline src/domain/books/BooksPage.tsx:30]: This is OK.
But a general rule of thumb is to always have the "success" case at the end. In this case, booksList should be returned at last and before that all the guard clauses are placed.

Another thing, just use If and return. No need for else-if and else blocks. They are redundant.

**Comment by atmeshwar-classic** [comment]
I am closing this PR as the assessment is complete.
@ansaf-1246 Please look at the pointers I have given to you.

**Pull request closed without merging**